jb/20210103/split_of_child_share_fixed.py

```diff
@@ -0,0 +1,362 @@
+                              (person2.child * person2.weight).sum())
+    new_total_adult_poor = ((person2.adult * person2.weight * 
+                             new_poor_all_ubi).sum())
+    new_adult_poverty_rate = ((new_total_adult_poor)/
+                              (person2.adult * person2.weight).sum())
+    new_total_poor_all_ubi = (new_poor_all_ubi * person2.weight).sum()
+    new_poverty_rate_all_ubi = new_total_poor_all_ubi/person2.weight.sum()
+    ubi_all_overall.append(new_poverty_rate_all_ubi)
+    ubi_all_child.append(new_child_poverty_rate)
+    ubi_all_adults.append(new_adult_poverty_rate)
+    
+spending_data = []
+for spending in range(0, 1001, 50):
+    spending = spending/100
+    spending_data.append(spending)
+    
+### ANALYSIS ###
+
+# Create a DataFrame grouped by each plans impact on the overall poverty rate. 
+overall = {'spending_in_billions': spending_data,
+                       'child_allowance': child_allowance_overall,
+                       'adult_ubi': ubi_adults_overall,
+                       'all_ubi': ubi_all_overall}
+                    
+overall_df = pd.DataFrame(overall)
+overall_df = pd.DataFrame(overall_df).round(3)
+
+# Create a DataFrame grouped by each plans impact on the child poverty rate.
+child = {'spending_in_billions': spending_data,
+         'child_allowance': child_allowance_child,
+         'adult_ubi': ubi_adults_child,
+         'all_ubi': ubi_all_child}
+                    
+child_df = pd.DataFrame(child)
+child_df = pd.DataFrame(child_df).round(3)
+
+
+# Create a DataFrame grouped by each plans impact on the adult poverty rate.
+adult = {'spending_in_billions': spending_data,
+         'child_allowance': child_allowance_adults,
+         'adult_ubi': ubi_adults_adults,
+         'all_ubi': ubi_all_adults}
+                    
+adult_df = pd.DataFrame(adult)
+adult_df = pd.DataFrame(adult_df).round(3)
+
+
+# Join different programs together for plotly.
+program = (pd.melt(overall_df, 'spending_in_billions', 
+                   var_name='ubi_type',value_name='poverty_rate'))
+
+def melt_dict(d):
+  """ produce long version of data frame represented by dictionary (d).
+  
+  Arguments
+  d: Dictionary where each element represents a differnt UBI type and spending levels and the poverty impacts.
+  
+  Returns
+  DataFrame where every row is the combination of UBI type and spending level.
+  """
+  df = pd.DataFrame(d).round(3) * 100
+  program = pd.melt(df, 'spending_in_billions', var_name='ubi_type',value_name='poverty_rate')
+  program['ubi_type'] = program.ubi_type.map({'child_allowance': 'Child allowance',
+                                      'adult_ubi': 'Adult UBI',
+                                      'all_ubi': 'All UBI'})
+  return program
+
+program_overall = melt_dict(overall)
+program_child = melt_dict(child)
+program_adult = melt_dict(adult)
+
+def line_graph(df, x, y, color, title, xaxis_title, yaxis_title):
```

Comment:
remove

Reply:
deleting all of this, old code from july post that Nate was using for reference